.github/workflows/README.md

```diff
@@ -0,0 +1,50 @@
+## CI/CD
+
+There are 5 main stages that run on GitHub actions:
+
+* Build
+* Unit Test
+* Integration Test
+* Release
+
+There are some other stages that run for every push on the main branches:
+
+* [Microbenchmark](./microbenchmark.yml)
+* [Snapshoty](./snapshoty.yml)
+* [Sync-branches](./sync-branches.yml)
+
+### Scenarios
+
+* Tests should be triggered on branch, tag and PR basis.
+* Commits that are only affecting the docs files should not trigger any test or similar stages that are not required.
+* Automated release in the CI gets triggered through a GitHub workflow.
+* **This is not the case yet**, but if Github secrets are required then Pull Requests from forked repositories won't run any build accessing those secrets. If needed, then create a feature branch.
+
+### How to interact with the CI?
+
+#### On a PR basis
+
+Once a PR has been opened then there are two different ways you can trigger builds in the CI:
+
+1. Commit based
+1. UI based, any Elasticians can force a build through the GitHub UI
+
+#### Branches
+
+Every time there is a merge to main or any release branches the whole workflow will compile and test on Linux and Windows.
```

Comment:
For "release branches" here, do you mean the `1.x` or hypothetical `2.x` we use to update documentation ? Our releases are happening on `main` branch for now.

Reply:
> do you mean the 1.x or hypothetical 2.x we use to update documentation ?

Correct, I didn't want to be too accurate about what release branches mean in this particular case, but keeping a bit generic.

If that's the case, I'll remove the `release` but keeping `branches` so in case there are feature branches in this project

Reply:
```suggestion
Every time there is a merge to main or any branches the whole workflow will compile and test on Linux and Windows.
```